compiler/src/dotty/tools/dotc/transform/CheckUnused.scala

```diff
@@ -661,8 +661,10 @@ object CheckUnused:
         val simpleSelections = qual.tpe.member(sym.name).alternatives
         val typeSelections = sels.flatMap(n => qual.tpe.member(n.name.toTypeName).alternatives)
         val termSelections = sels.flatMap(n => qual.tpe.member(n.name.toTermName).alternatives)
+        val sameTermPath = qual.isTerm && sym.exists && sym.owner.isType && qual.tpe.typeSymbol == sym.owner.asType
         val selectionsToDealias = typeSelections ::: termSelections
-        val qualHasSymbol = simpleSelections.map(_.symbol).contains(sym) || (simpleSelections ::: selectionsToDealias).map(_.symbol).map(dealias).contains(dealiasedSym)
+        def renamedSelection = if(sameTermPath) then sels.find(sel => sel.imported.name == sym.name) else None
```

Comment:
`renamedSelection` will be computed multiple times, it shouldn't matter that much, but maybe `val` makes as much sense here?
Also the formatting of this `if` seems a bit off.

Reply:
A lazy val would make more sense I think. It's last in the chain of `or`s, so I wanted to avoid calculating it eagerly.

Reply:
huh, the tests failed after changing def => lazy val, that's quite interesting